aten/src/ATen/core/jit_type.h

```diff
@@ -824,6 +832,9 @@ struct CAFFE2_API NoneType : public Type {
   }
 
   bool isSubtypeOf(const TypePtr rhs) const override {
+    if(auto rhs_ = rhs->cast<OptionalType>()) {
```

Comment:
NoneType is no longer a Subtype of any optional type, see the discussion in #13959